security-oauth2/src/test/groovy/io/micronaut/security/oauth2/DefaultSecureEndpointTest.java

```diff
@@ -0,0 +1,43 @@
+package io.micronaut.security.oauth2;
+
+import io.micronaut.security.oauth2.endpoint.AuthenticationMethod;
+import io.micronaut.security.oauth2.endpoint.DefaultSecureEndpoint;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class DefaultSecureEndpointTest {
+    private static final List<String> METHODS = List.of("client_secret_basic",
+            "client_secret_post",
+            "client_secret_jwt",
+            "private_key_jwt",
+            "tls_client_auth",
+            "self_signed_tls_client_auth");
+
+    @Test
+    void testGetAuthenticationMethodsSupported() {
```

Comment:
I'm not a big fan of mock tests. Can this be easily tested using the bean context?

Reply:
`DefaultSecureEndpoint` is not a bean. I added this test to fix sonar coverage notice. I am looking if I can add an extra test using the context.